File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/60968-Visual Studio 2010 SDK Samples/VSSDK IDE Sample Editor with Toolbox/C#/UnitTests/EditorPackageTest.cs
/***************************************************************************

Copyright (c) Microsoft Corporation. All rights reserved.
This code is licensed under the Visual Studio SDK license terms.
THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

***************************************************************************/

using System;
using System.Text;
using Microsoft.VisualStudio;
using System.Collections.Generic;
using OleInterop = Microsoft.VisualStudio.OLE.Interop;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Microsoft.VisualStudio.Shell.Interop;
using Microsoft.VsSDK.UnitTestLibrary;

namespace Microsoft.Samples.VisualStudio.IDE.EditorWithToolbox.UnitTests
{
    /// <summary>
    ///This is a test class for Microsoft.Samples.VisualStudio.IDE.EditorWithToolbox.EditorPackage and is intended
    ///to contain all Microsoft.Samples.VisualStudio.IDE.EditorWithToolbox.EditorPackage Unit Tests
    ///</summary>
    [TestClass()]
    public class EditorPackageTest : IDisposable
    {
        #region Fields

        private TestContext testContextInstance;
        private EditorPackage editorPackage;

        #endregion Fields

        #region Properties
        /// <summary>
        ///Gets or sets the test context which provides
        ///information about and functionality for the current test run.
        ///</summary>
        public TestContext TestContext
        {
            get
            {
                return testContextInstance;
            }
            set
            {
                testContextInstance = value;
            }
        }
        #endregion Properties

        #region Initialization && Cleanup
        /// <summary>
        /// Runs before the test to allocate and configure resources needed 
        /// by all tests in the test class.
        /// </summary>
        [TestInitialize()]
        public void EditorPackageTestInitialize()
        {
            editorPackage = new EditorPackage();
        }
        /// <summary>
        /// Runs after the test has run and to free resources obtained 
        /// by all the tests in the test class.
        /// </summary>
        [TestCleanup()]
        public void EditorPackageTestCleanup()
        {
            editorPackage = null;
        }

        #endregion

        #region IDisposable Pattern implementation
        /// <summary>
        /// Implement IDisposable.
        /// </summary>
        public void Dispose()
        {
            Dispose(true);
        }
        /// <summary> 
        /// Clean up any resources being used.
        /// </summary>
        protected virtual void Dispose(bool disposing)
        {
            if (disposing)
            {
                if (editorPackage != null)
                {
                    editorPackage.Dispose();
                    editorPackage = null;
                }
                GC.SuppressFinalize(this);
            }
        }
        #endregion

        #region IDisposable tests
        /// <summary>
        ///A test for Dispose (bool)
        ///</summary>
        [TestMethod()]
        public void DisposeTest()
        {
            EditorPackage target = editorPackage;
            Microsoft_Samples_VisualStudio_IDE_EditorWithToolbox_EditorPackageAccessor accessor = new Microsoft_Samples_VisualStudio_IDE_EditorWithToolbox_EditorPackageAccessor(target);
            accessor.editorFactory = new EditorFactory();
            target.Dispose();
            Assert.IsNull(accessor.editorFactory, "Internal Editor Factory object was not disposed properly.");
        }
        #endregion IDisposable tests

        [TestMethod()]
        public void SetSite()
        {
            // Create the package
            IVsPackage package = new EditorPackage() as IVsPackage;
            Assert.IsNotNull(package, "The object does not implement IVsPackage");

            // Create a basic service provider
            OleServiceProvider serviceProvider = OleServiceProvider.CreateOleServiceProviderWithBasicServices();

            // Add site support to register editor factory
            BaseMock registerEditor = MockRegisterEditors.GetRegisterEditorsInstance();
            serviceProvider.AddService(typeof(SVsRegisterEditors), registerEditor, false);

            // Site the package
            Assert.AreEqual(0, package.SetSite(serviceProvider), "SetSite did not return S_OK");
        }
    }
}

File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/60968-Visual Studio 2010 SDK Samples/VSSDK IDE Sample Editor with Toolbox/VB/TDD/EditorPackageTest.vb
'**************************************************************************

'Copyright (c) Microsoft Corporation. All rights reserved.
'This code is licensed under the Visual Studio SDK license terms.
'THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
'ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
'IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
'PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

'**************************************************************************


Imports Microsoft.VisualBasic
Imports System
Imports System.Text
Imports Microsoft.VisualStudio
Imports System.Collections.Generic
Imports OleInterop = Microsoft.VisualStudio.OLE.Interop
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports Microsoft.VisualStudio.Shell.Interop
Imports Microsoft.VsSDK.UnitTestLibrary

Namespace Microsoft.Samples.VisualStudio.IDE.EditorWithToolbox.UnitTests
	''' <summary>
    ''' This is a test class for Microsoft.Samples.VisualStudio.IDE.EditorWithToolbox.EditorPackage and is intended
    ''' to contain all Microsoft.Samples.VisualStudio.IDE.EditorWithToolbox.EditorPackage Unit Tests.
    ''' </summary>
	<TestClass()> _
	Public Class EditorPackageTest
		Implements IDisposable
		#Region "Fields"

		Private testContextInstance As TestContext
		Private editorPackage As EditorPackage

		#End Region
		#Region "Properties"
		''' <summary>
        ''' Gets or sets the test context which provides
        ''' information about and functionality for the current test run.
		'''</summary>
		Public Property TestContext() As TestContext
			Get
				Return testContextInstance
			End Get
			Set(ByVal value As TestContext)
				testContextInstance = value
			End Set
		End Property
#End Region

		#Region "Initialization && Cleanup"
		''' <summary>
		''' Runs before the test to allocate and configure resources needed 
		''' by all tests in the test class.
		''' </summary>
		<TestInitialize()> _
		Public Sub EditorPackageTestInitialize()
			editorPackage = New EditorPackage()
		End Sub
		''' <summary>
		''' Runs after the test has run and to free resources obtained 
		''' by all the tests in the test class.
		''' </summary>
		<TestCleanup()> _
		Public Sub EditorPackageTestCleanup()
			editorPackage = Nothing
		End Sub

		#End Region

		#Region "IDisposable Pattern implementation"
		''' <summary>
		''' Implement IDisposable.
		''' </summary>
		Public Sub Dispose() Implements IDisposable.Dispose
			Dispose(True)
        End Sub

		''' <summary> 
		''' Clean up any resources being used.
		''' </summary>
		Protected Overridable Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
                If editorPackage IsNot Nothing Then
                    editorPackage.Dispose()
                    editorPackage = Nothing
                End If
				GC.SuppressFinalize(Me)
			End If
		End Sub
		#End Region

		#Region "IDisposable tests"
		''' <summary>
        ''' A test for Dispose (bool).
        ''' </summary>
		<TestMethod()> _
		Public Sub DisposeTest()
			Dim target As EditorPackage = editorPackage
            Dim accessor As New Microsoft_Samples_VisualStudio_IDE_EditorWithToolbox_EditorPackageAccessor(target)
			accessor.editorFactory = New EditorFactory()
			target.Dispose()
			Assert.IsNull(accessor.editorFactory, "Internal Editor Factory object was not disposed properly.")
		End Sub
#End Region

		<TestMethod()> _
		Public Sub SetSite()
            ' Create the package.
            Dim package As IVsPackage = TryCast(New EditorPackage(), IVsPackage)
			Assert.IsNotNull(package, "The object does not implement IVsPackage")

            ' Create a basic service provider.
			Dim serviceProvider As OleServiceProvider = OleServiceProvider.CreateOleServiceProviderWithBasicServices()

            ' Add site support to register editor factory.
			Dim registerEditor As BaseMock = MockRegisterEditors.GetRegisterEditorsInstance()
			serviceProvider.AddService(GetType(SVsRegisterEditors), registerEditor, False)

            ' Site the package.
			Assert.AreEqual(0, package.SetSite(serviceProvider), "SetSite did not return S_OK")
		End Sub
	End Class
End Namespace
